### ISSUE TITLE
Unclear what training functions should instantiate

### ISSUE THREAD

**Issue by worldveil** [ray-project/ray#22079]
It's not really clear to a user in the [Ray Train porting documentation](https://docs.ray.io/en/latest/train/user_guide.html#run-training-function) what *actually should be created* in the training function, or what is best practice. 

The diff breakdown is good, but sort of incomplete:

<img width="772" alt="Screen Shot 2022-02-03 at 11 53 32 AM" src="https://user-images.githubusercontent.com/1045250/152409751-18b5b201-6378-471c-b4a2-47ac3e97b18d.png">

Broadly there are few main components we should detail for instantiation:

- torch datasets? (no idea)
- models? (yes?)
- training loop? (yes)
- optimizers/loss? (no idea)

One of the points of using Ray is making it easier for developers to not think about the complexities of shared state in a distributed cluster setting. We need to walk them through where they should create objects (state) and how Ray Train handles it.

**Comment by worldveil** [comment]
Can we add the list of things a user must create inside (or whether should be outside) for a typical torch/TF script to docs [here](https://docs.ray.io/en/latest/train/user_guide.html#update-training-function)? 

between : 

- torch datasets? 
- models?
- training loop? 
- optimizers/loss? 
- ...anything else I missed

?

**Comment by dstrodtman** [comment]
Closing as obsolete. The page or content referenced here has been removed or substantially rewritten since the issue was filed, so the original report no longer applies. If a similar gap exists today, please open a fresh issue with a link to the current page.

Context: docs-team triage of the OSS docs issue backlog.
